trio/_tests/test_threads.py

```diff
@@ -500,55 +541,57 @@ async def run_thread(event):
         c.total_tokens = orig_total_tokens
 
 
-async def test_run_in_worker_thread_custom_limiter():
+async def test_run_in_worker_thread_custom_limiter() -> None:
     # Basically just checking that we only call acquire_on_behalf_of and
     # release_on_behalf_of, since that's part of our documented API.
     record = []
 
     class CustomLimiter:
-        async def acquire_on_behalf_of(self, borrower):
+        async def acquire_on_behalf_of(self, borrower: Task) -> None:
             record.append("acquire")
             self._borrower = borrower
 
-        def release_on_behalf_of(self, borrower):
+        def release_on_behalf_of(self, borrower: Task) -> None:
             record.append("release")
             assert borrower == self._borrower
 
-    await to_thread_run_sync(lambda: None, limiter=CustomLimiter())
+    # TODO: should CapacityLimiter have an abc or protocol so users can modify it?
+    # because currently it's `final` so writing code like this is not allowed.
+    await to_thread_run_sync(lambda: None, limiter=CustomLimiter())  # type: ignore[arg-type]
     assert record == ["acquire", "release"]
 
 
-async def test_run_in_worker_thread_limiter_error():
+async def test_run_in_worker_thread_limiter_error() -> None:
     record = []
 
     class BadCapacityLimiter:
-        async def acquire_on_behalf_of(self, borrower):
+        async def acquire_on_behalf_of(self, borrower: Task) -> None:
             record.append("acquire")
 
-        def release_on_behalf_of(self, borrower):
+        def release_on_behalf_of(self, borrower: Task) -> NoReturn:
             record.append("release")
             raise ValueError
 
     bs = BadCapacityLimiter()
 
     with pytest.raises(ValueError) as excinfo:
-        await to_thread_run_sync(lambda: None, limiter=bs)
+        await to_thread_run_sync(lambda: None, limiter=bs)  # type: ignore[arg-type]
     assert excinfo.value.__context__ is None
     assert record == ["acquire", "release"]
     record = []
 
     # If the original function raised an error, then the semaphore error
     # chains with it
-    d = {}
+    d: dict[str, object] = {}
     with pytest.raises(ValueError) as excinfo:
-        await to_thread_run_sync(lambda: d["x"], limiter=bs)
+        await to_thread_run_sync(lambda: d["x"], limiter=bs)  # type: ignore[arg-type]
     assert isinstance(excinfo.value.__context__, KeyError)
     assert record == ["acquire", "release"]
 
 
-async def test_run_in_worker_thread_fail_to_spawn(monkeypatch):
+async def test_run_in_worker_thread_fail_to_spawn(monkeypatch: MonkeyPatch) -> None:
     # Test the unlikely but possible case where trying to spawn a thread fails
-    def bad_start(self, *args):
+    def bad_start(self: object, *args: object) -> NoReturn:
         raise RuntimeError("the engines canna take it captain")
```

Comment:
Unrelated to the types side of things but noticed this, could we change `canna` -> `cannot`

Reply:
lol, I suspect it might be an intentional joke - objectively the error message is pretty unhelpful. I vote to keep it :pirate_flag:

Reply:
It's a Star Trek reference indeed.